libs/src/WebAudiusLibs.ts

```diff
@@ -538,11 +538,26 @@ export class AudiusLibs {
     /** Creator Node */
     if (this.creatorNodeConfig) {
       const currentUser = this.userStateManager.getCurrentUser()
-      const creatorNodeEndpoint = currentUser
+      let creatorNodeEndpoint = currentUser
         ? CreatorNode.getPrimary(currentUser.creator_node_endpoint) ??
           this.creatorNodeConfig.fallbackUrl
         : this.creatorNodeConfig.fallbackUrl
 
+      // Use rendezvous to select creatorNodeEndpoint for v2 users
+      if (currentUser?.is_storage_v2 && this.ethContracts) {
```

Comment:
Would probably require passing a bunch of stuff in but would it be better to do this logic in the `CreatorNode` constructor? Especially since we are having to duplicate it for AudiusLibs/NativeAudiusLibs/WebAudiusLibs

Reply:
so the main blocker with that is that it requires an async call, so it can't be in the constructor. I tried to put it in `CreatorNode`'s init function which is called by AudiusLibs/NativeAudiusLibs/WebAudiusLibs, but then the problem is it needs ServiceProvider. ServiceProvider can't be given to CreatorNode because ServiceProvider has all the services including CreatorNode :/ wdyt though maybe I'm missing a better solution?